testsuite/scripts/src/test/java/org/wildfly/test/scripts/AppClientScriptTestCase.java

```diff
@@ -22,20 +22,29 @@ public AppClientScriptTestCase() {
 
     @Override
     void testScript(final ScriptProcess script) throws InterruptedException, TimeoutException, IOException {
+        // First check the standard script
         script.start(MAVEN_JAVA_OPTS, "-v");
-        Assert.assertNotNull("The process is null and may have failed to start.", script);
+        testScript(script, 0);
 
-        validateProcess(script);
+        // Test with the security manager enabled
+        script.start(MAVEN_JAVA_OPTS, "-v", "-secmgr");
+        testScript(script, jvmVersion() >= 17 ? 4 : 0);
```

Comment:
Don't update this now, as I'll merge this today if I can, but AssumeTestGroupUtil.isJDKVersionAfter(16) or isJDKVersionBefore(17) would handle this.

Reply:
Ah, right. I should have checked that. The `isJDKverserionBefore(17)` would be what we need here. The extra lines are warnings I should have probably documented in a comment.